src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```diff
@@ -92,31 +123,100 @@ public override Task SendUsersAsync(IReadOnlyList<string> userIds, string method
+            }
+
+            var api = _restApiProvider.GetUserGroupManagementEndpoint(userId, groupName);
+            var request = BuildRequest(api, HttpMethod.Delete, null, null, cancellationToken);
+            return CallRestApiAsync(request);
         }
 
-        private HttpRequestMessage GenerateHttpRequest(string url, PayloadMessage payload, string tokenString, HttpMethod httpMethod)
+        private static HttpRequestMessage GenerateHttpRequest(string url, PayloadMessage payload, string tokenString, HttpMethod httpMethod)
         {
             var request = new HttpRequestMessage(httpMethod, url);
             request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokenString);
             request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             request.Content = new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json");
             return request;
         }
+
+        private static void ThrowExceptionOnResponseFailure(Exception innerException, HttpStatusCode? statusCode, string requestUri, string detail = null)
+        {
+            switch (statusCode)
+            {
+                case HttpStatusCode.BadRequest:
+                    {
+                        throw new AzureSignalRInvalidArgumentException(requestUri, innerException, detail);
+                    }
+                case HttpStatusCode.Unauthorized:
+                    {
+                        throw new AzureSignalRUnauthorizedException(requestUri, innerException);
+                    }
+                case HttpStatusCode.NotFound:
+                    {
+                        throw new AzureSignalRInaccessibleEndpointException(requestUri, innerException);
+                    }
+                default:
+                    {
+                        throw new AzureSignalRRuntimeException(requestUri, innerException);
+                    }
+            }
+        }
+
+        private static HttpRequestMessage BuildRequest(RestApiEndpoint endpoint, HttpMethod httpMethod, string methodName = null, object[] args = null, CancellationToken cancellationToken = default)
+        {
+            var payload = httpMethod == HttpMethod.Post ? new PayloadMessage { Target = methodName, Arguments = args } : null;
+            return GenerateHttpRequest(endpoint.Audience, payload, endpoint.Token, HttpMethod.Post);
+        }
+
+        private static async Task CallRestApiAsync(HttpRequestMessage request)
+        {
+            var httpClient = HttpClientFactory.CreateClient();
+            HttpResponseMessage response = null;
+            string detail = "";
+
+            try
+            {
+                response = await httpClient.SendAsync(request);
+            }
+            catch (HttpRequestException ex)
+            {
+                ThrowExceptionOnResponseFailure(ex, HttpStatusCode.NotFound, request.RequestUri.ToString());
```

Comment:
I would rather directly throw instead of pass in a NotFound status code.

Reply:
@chenkennt updated